src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -118,6 +116,20 @@ public virtual bool CanValidateToken
             get { return true; }
         }
 
+        private int CountTokenSegments(string token)
```

Comment:
nit: callers of this method checks that this is not null or empty. Code can be moved around and those checks might disappear, perhaps there should be a nullorempty check here as well returning count == 0 for that case? Or alternatively throw an exception. Nit as this is not an issue in the current code...

Reply:
I've added null check and return 0 in that case. I find it hard to move those checks here as they're serving different purposes in each case, but I'm open to advice.